### ISSUE TITLE
SQLCipher Compatibility Issue: file is not a database

### ISSUE THREAD

**Issue by alvisjosh** [sqlcipher/android-database-sqlcipher#398]
We have been using SQLCipher 3.4.2 jar.

Recently, we updated the library to the latest android-database-sqlcipher 3.5.9, and our Android app was hit with the following error.

Error: file is not a database: , while compiling: select count(*) from sqlite_master;

### Expected Behavior
We are expecting both SQLCipher 3.4.2 and 3.5.9 to open our sqlite db without any error.

### Actual Behavior
App crashes with error: file is not a database: , while compiling: select count(*) from sqlite_master;

Any help or advice would be greatly appreciated. Thank you.

**Comment by developernotes** [comment]
Hello @alvisjosh 

Can you tell us a bit more about both the key material used on the database you are trying to access (i.e., how was it generated), and any adjustments to the runtime configuration options you may have made? Are you certain the correct password is being provided?

**Comment by alvisjosh** [comment]
Yeah, I am certain the password was correct. I have done another test.

Downloaded both SQLiteManager and DB Browser for SQLite. DB Browser for SQLite refused to open the db with the correct password, but I was able to open my db using SQLiteManager.

Furthermore, if I used SQLiteManager to change the password of the db once, and I will be able to use DB Browser for SQLite to open the db with a the new password.

**Comment by developernotes** [comment]
Hello @alvisjosh 

Can you tell us a bit more about both the key material used on the database you are trying to access (i.e., how was it generated), and any adjustments to the runtime configuration options you may have made?

Also, can you provide a sample database and password to further analyze the behavior you are experiencing?

**Comment by takyeyu** [comment]
Hi @developernotes , I have the same problem, but only in device with android 9, and it is fine with version <= android 8.1. attached please find the simple db and the password is Qa$8j%nttJ3*m^f*SQEj^b$c&mZ31_@1!AbBQ@y^x&@dA62&LG

I create the db "DB Browser for SQLite" with password settings (passphrase and page size is 1024), and it is fine to use DB Browser to query the db

Lib version: android-database-sqlcipher 3.5.9

[v4_0.db.zip](https://github.com/sqlcipher/android-database-sqlcipher/files/2494506/v4_0.db.zip)

**Comment by developernotes** [comment]
Hello @takyeyu 

I was unable to open this database file in either the SQLCipher command line shell, or DB Browser for SQLite. If you are having trouble with a specific scenario, you might consider trying out the SQLCipher for Android [test suite](https://github.com/sqlcipher/sqlcipher-android-tests) to help troubleshoot your scenario. 

Are you able to recreate the database file using the SQLCipher command line shell and see if you have any issues opening it when using the Android library?

**Comment by JellyPaul** [comment]
Hi, I have the same problem in multiprocessing app. I am sure that I use the same password, It occurred occasionally.
Lib version: 3.5.4

**Comment by sjlombardo** [comment]
@JellyPaul you should try to update to the latest stable release.
